### PR TITLE
[rpc][jit] support rpc_sync in TorchScript

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43046 [rpc][jit] support remote call in TorchScript
* **#43043 [rpc][jit] support rpc_sync in TorchScript**
* #43042 [rpc] rename some functions

This add the support for rpc_sync in TorchScript in a way similar to
rpc_async

Differential Revision: [D23252039](https://our.internmc.facebook.com/intern/diff/D23252039)

### CODE REVIEW

File: torch/csrc/jit/python/python_sugared_value.cpp [hunk 1]
```diff
@@ -886,6 +886,11 @@ std::shared_ptr<SugaredValue> toSugaredValue(
       obj.ptr() ==
       py::module::import("torch.distributed.rpc").attr("rpc_async").ptr()) {
     return SpecialFormValue::create(prim::rpc_async);
+  } else if (
+      // RPC module is only avaialble  when build flag "USE_DISTRIBUTED" is on.
```
Comment (mrshenli) [inline torch/csrc/jit/python/python_sugared_value.cpp:890]: redundant space after "available"

File: torch/csrc/jit/python/python_sugared_value.cpp [hunk 2]
```diff
@@ -886,6 +886,11 @@ std::shared_ptr<SugaredValue> toSugaredValue(
...
+      obj.ptr() ==
+      py::module::import("torch.distributed.rpc").attr("rpc_sync").ptr()) {
```
Comment (mrshenli) [inline torch/csrc/jit/python/python_sugared_value.cpp:892]: is the indent here intentional?

File: torch/csrc/jit/runtime/register_distributed_ops.cpp [hunk 3]
```diff
@@ -26,6 +26,136 @@ static auto workerInfo =
...
+  } else if (rpc_op == "rpc_sync") {
+    // Send RPC request.
+    auto futureIValuePtr = dist_rpc::rpcTorchscript(
+        dstWorkerNameStr,
+        qualifiedName,
+        functionSchema,
+        userCallableStack,
+        rpcTimeout);
+    futureIValuePtr->wait();
+    auto res = futureIValuePtr->value();
```
Comment (mrshenli) [inline torch/csrc/jit/runtime/register_distributed_ops.cpp:149]: These could throw error. Is that OK?

**Review by mrshenli — APPROVED** [review]
LGTM! Thanks for adding this!

It will be great if we can also cover the case where user function throws, but it can be added in follow up PRs.

File: torch/csrc/jit/python/python_sugared_value.cpp (hunk 2 above)
Comment (wanchaol) [reply torch/csrc/jit/python/python_sugared_value.cpp:891]: it's formatted by clang-format (and stay aligned with the above rpc_async op)

File: torch/csrc/jit/runtime/register_distributed_ops.cpp (hunk 3 above)
Comment (wanchaol) [reply torch/csrc/jit/runtime/register_distributed_ops.cpp]: good catch! handled the error.

**Pull request closed without merging**